src/torchcodec/_core/BetaCudaDeviceInterface.cpp

```diff
@@ -216,12 +216,11 @@ BetaCudaDeviceInterface::~BetaCudaDeviceInterface() {
     // unclear.
     flush();
     unmapPreviousFrame();
-    NVDECCache::getCache(device_.index())
-        .returnDecoder(&videoFormat_, std::move(decoder_));
+    NVDECCache::getCache(device_).returnDecoder(
+        &videoFormat_, std::move(decoder_));
```

Comment:
Above and everywhere else in the PR, anything related to "device index" relates to the creation of `getDeviceIndex()` which is now in `CUDACommon.cpp`, and replaces the old (buggy!) `getNonNegativeDeviceIndex`.